src/Compiler/Checking/NameResolution.fs

```diff
@@ -4346,6 +4332,54 @@ let rec private EntityRefContainsSomethingAccessible (ncenv: NameResolver) m ad
         let submodref = modref.NestedTyconRef submod
         EntityRefContainsSomethingAccessible ncenv m ad submodref))
 
+let GetVisibleNamespacesAndModulesAtPoint (ncenv: NameResolver) (nenv: NameResolutionEnv) fullyQualified m ad =
+    protectAssemblyExploration [] (fun () ->
+        let items =
+            nenv.ModulesAndNamespaces fullyQualified
+            |> NameMultiMap.range
+
+        if isNil items then
+            []
+        else
+            let ilTyconNames =
+                let hashSet = HashSet()
+                for tyconRef in nenv.TyconsByAccessNames(fullyQualified).Values do
+                    if tyconRef.IsILTycon then
+                        hashSet.Add tyconRef.DisplayName |> ignore
+                hashSet
+
+            items
+            |> List.filter (fun x ->
+                 let demangledName = x.DemangledModuleOrNamespaceName
+                 IsInterestingModuleName demangledName && notFakeContainerModule ilTyconNames demangledName
+                 && EntityRefContainsSomethingAccessible ncenv m ad  x
+                 && not (IsTyconUnseen ad ncenv.g ncenv.amap m x)))
+
+let GetAccessibleSubModules g (ncenv: NameResolver) (modref: ModuleOrNamespaceRef) m ad =
```

Comment:
For future reference - it's technically a "submodule" (one word), so no need to capitalize M.

Reply:
(I know it was like this before :D)